test/e2e/cluster_logging_gcl.go

```diff
@@ -0,0 +1,140 @@
+/*
+Copyright 2015 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"os/exec"
+	"strconv"
+	"strings"
+	"time"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/util/json"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = framework.KubeDescribe("Cluster level logging using GCL", func() {
+	f := framework.NewDefaultFramework("gcl-logging")
+
+	BeforeEach(func() {
+		// TODO (crassirostris): Expand to GKE once the test is stable
+		framework.SkipUnlessProviderIs("gce")
+	})
+
+	It("should check that logs from containers are ingested in GCL", func() {
+		synthLoggerPodName := f.Namespace.Name + "-synthlogger"
+
+		By("Creating synthetic logger")
+		createSynthLogger(f, synthLoggerPodName, expectedLinesCount)
+		defer f.PodClient().Delete(synthLoggerPodName, &api.DeleteOptions{})
+
+		By("Waiting for logs to ingest")
+		totalMissing := expectedLinesCount
+		for start := time.Now(); totalMissing > 0 && time.Since(start) < ingestionTimeout; time.Sleep(25 * time.Second) {
+			var err error
+			totalMissing, err = getMissingLinesCountGcl(synthLoggerPodName, expectedLinesCount)
+			if err != nil {
+				framework.Logf("Failed to get missing lines count due to %v", err)
+				totalMissing = expectedLinesCount
+			} else if totalMissing > 0 {
+				framework.Logf("Still missing %d lines", totalMissing)
+			}
+		}
+
+		Expect(totalMissing).To(Equal(0), "Some log lines are still missing")
+	})
+})
+
+func getMissingLinesCountGcl(podName string, expectedCount int) (int, error) {
+	gclFilter := fmt.Sprintf("resource.labels.pod_id:%s", podName)
```

Comment:
Is pod name exported as pod_id? If yes this might be confusing, because pod has also id.